test.sh

```diff
@@ -1,80 +1,132 @@
+    echo "-s run in strict mode (sets \$SECURITY=\"strict\")"
+    echo "Cluster must be created and \$CLUSTER_URL set"
+    echo "AWS credentials must exist in the variables:"
+    echo "      \$AWS_ACCESS_KEY_ID"
+    echo "      \$AWS_SECRET_ACCESS_KEY"
+    echo "Azure tests will run if thses variables are set:"
+    echo "      \$AWS_ACCESS_KEY_ID"
+    echo "      \$AWS_SECRET_ACCESS_KEY"
+    echo "      \$AZURE_CLIENT_ID"
+    echo "      \$AZURE_CLIENT_SECRET"
+    echo "      \$AZURE_TENANT_ID"
+    echo "      \$AZURE_STORAGE_ACCOUNT"
+    echo "      \$AZURE_STORAGE_KEY"
+    echo "  (changes the -m default to \"sanity\")"
+    echo "Set \$STUB_UNIVERSE_URL to bypass build"
+    echo "  (invalid when building all frameworks)"
+    echo "Current frameworks:"
+    for framework in $FRAMEWORK_LIST; do
+        echo "       $framework"
+    done
+    exit 1
+}
 
-        if [ ! -f "$UNIVERSE_URL_PATH" ]; then
-            echo "Missing universe URL file: $UNIVERSE_URL_PATH"
-            exit 1
-        fi
-        export STUB_UNIVERSE_URL=$(cat $UNIVERSE_URL_PATH)
-        rm -f $UNIVERSE_URL_PATH
-        echo "Built/uploaded stub universe: $STUB_UNIVERSE_URL"
-    else
-        echo "Using provided STUB_UNIVERSE_URL: $STUB_UNIVERSE_URL"
-    fi
+if [ "$#" -eq "0" ]; then
+    usage
+fi
 
-    echo Security: $SECURITY
-    if [ "$SECURITY" = "strict" ]; then
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root ${framework}-role
-        # include foldered roles (tests exercise with /test/integration/svcname):
-        ${REPO_ROOT_DIR}/tools/setup_permissions.sh root test__integration__${framework}-role
-    fi
+if [ -z "$CLUSTER_URL" ]; then
+    echo "Cluster not found. Create and configure one then set \$CLUSTER_URL."
+    exit 1
+fi
 
-    # Run shakedown tests in framework directory:
-    TEST_GITHUB_LABEL="${framework}" ${REPO_ROOT_DIR}/tools/run_tests.py shakedown ${FRAMEWORK_DIR}/tests/
-}
+if [ -z "$AWS_ACCESS_KEY_ID" -o -z "$AWS_SECRET_ACCESS_KEY" ]; then
+    echo "AWS credentials not found (\$AWS_ACCESS_KEY_ID and \$AWS_SECRET_ACCESS_KEY)."
+    exit 1
+fi
 
-echo "Beginning integration tests at "`date`
+security="permissive"
+pytest_m="sanity and not azure"
+pytest_k=""
+azure_args=""
+ssh_path="${HOME}/.ssh/ccm.pem"
 
-REPO_ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# If AZURE variables are given, change default -m and prepare args for docker
+if [ -n "$AZURE_DEV_CLIENT_ID" -a -n "$AZURE_DEV_CLIENT_SECRET" -a \
+        -n "$AZURE_DEV_TENANT_ID" -a -n "$AZURE_DEV_STORAGE_ACCOUNT" -a \
+        -n "$AZURE_DEV_STORAGE_KEY" ]; then
+azure_args=$(cat<<EOFF
+    -e AZURE_CLIENT_ID=$AZURE_DEV_CLIENT_ID \
+    -e AZURE_CLIENT_SECRET=$AZURE_DEV_CLIENT_SECRET \
+    -e AZURE_TENANT_ID=$AZURE_DEV_TENANT_ID \
+    -e AZURE_STORAGE_ACCOUNT=$AZURE_DEV_STORAGE_ACCOUNT \
+    -e AZURE_STORAGE_KEY=$AZURE_DEV_STORAGE_KEY
+EOFF
+)
+    pytest_m="sanity"
+fi
 
-cd $REPO_ROOT_DIR
+while [[ $# -gt 1 ]]
```

Comment:
I don't see any long options here.  Why write custom getopts instead of just using the provided one?  I don't feel strongly enough to encourage extra work but I don't like the pattern.